src/ui/components/lobby/lobbyComponent.ts

```diff
@@ -28,77 +28,104 @@ Vue.partial("lobby-player-list", require("./partials/playerList.html"));
             // way we get notified of changes and update.
             settings: SettingServiceStatic,
         };
     }
-    
-    created() {        
+
+    created() {
         // Mainly for debugging. Redirects if there's no connection.
         if (!NetworkServiceStatic.currentConnection) {
-            this.$router.go("/loading");
+            this.$router.go("/login");
         }
-        
+
+        NetworkServiceStatic.on("host", isHost => {
+            this.isHost = isHost;
+        });
+
+        NetworkServiceStatic.on("start-championselect", () => {
+            this.$router.go("/championSelect");
+        });
+
         NetworkServiceStatic.on("chat", (timestamp, sender, message) => {
             if (this.messages.length && this.messages[this.messages.length - 1].sender === sender) {
                 this.messages[this.messages.length - 1].messages.push(message);
             } else {
                 this.messages.push({ timestamp, sender, messages: [message] });
             }
-            
+
             // Scroll down on the next tick if we aren't scrolled yet.
             // We scroll down next tick because by then the contents of the
             // chat will have been updated with the newly added message.
             const el = document.getElementById("chat-container");
             const isAttached = el.scrollTop + el.clientHeight + 1 >= el.scrollHeight;
-            
+
             this.$nextTick(() => {
                 // If we were attached before the element got bigger, scroll down.
                 // This effectively reattaches the scroll.
                 if (isAttached) {
                     el.scrollTop = el.scrollHeight;
                 }
-            });          
+            });
         });
+
+        NetworkServiceStatic.currentLobbyConnection.emit("host");
     }
-    
+
     playersInTeam(team: lobby.Team) {
         return this.lobby.players.filter(x => x.team === team);
     }
-    
+
     getOrNone(url = CDNService.getPath("assets/storeImages/layout/g-background_generic.jpg")) {
         return url;
     }
-    
+
     sendChat() {
         NetworkServiceStatic.sendMessage(this.message);
         this.message = "";
     }
-    
+
     joinTeam(team: lobby.Team) {
         NetworkServiceStatic.joinTeam(team);
     }
-    startGame(){
-        NetworkServiceStatic.startGame();
+
+    startGame() {
+        NetworkServiceStatic.startChampionSelect();
     }
-    
-    get playerSettings() {
-        return this.settings.playerSettings;
+
+    leave() {
+        NetworkServiceStatic.leaveLobby();
+        this.$router.go("/lobbies");
     }
-    
+
+    iconByUser(user) {
+        return StaticService.iconById(NetworkServiceStatic.userById(user.idServer).idIcon).iconURL;
+    }
+
     get hostSettings() {
         return this.settings.hostSettings;
     }
+
+    openLobbySettings() {
+        ModalComponent.present("lobby-settings", this.settings).then(c => {
+
+        });
```

Comment:
If you are going to do nothing with the resulting promise, might as well remove the `then` call.